src/coreclr/vm/assemblyspec.cpp

```diff
@@ -755,16 +726,16 @@ ICLRPrivBinder* AssemblySpec::GetBindingContextFromParentAssembly(AppDomain *pDo
 
     if (pParentAssemblyBinder != NULL)
     {
-        CLRPrivBinderCoreCLR *pTPABinder = pDomain->GetTPABinderContext();
-        if (AreSameBinderInstance(pTPABinder, pParentAssemblyBinder))
+        DefaultAssemblyBinder *pTPABinder = pDomain->GetTPABinderContext();
+        if (pTPABinder == pParentAssemblyBinder)
         {
             // If the parent assembly is a platform (TPA) assembly, then its binding context will always be the TPABinder context. In
             // such case, we will return the default context for binding to allow the bind to go
             // via the custom binder context, if it was overridden. If it was not overridden, then we will get the expected
             // TPABinder context anyways.
             //
             // Get the reference to the default binding context (this could be the TPABinder context or custom AssemblyLoadContext)
-            pParentAssemblyBinder = static_cast<ICLRPrivBinder*>(pDomain->GetTPABinderContext());
+            pParentAssemblyBinder = static_cast<AssemblyBinder*>(pDomain->GetTPABinderContext());
```

Comment:
Do we even need the specific cast here? This should cast implicitly just fine.

Reply:
Actually - we don't any of this now - if `pParentAssemblyBinder` is already the default binder - then this will assign the exact same value to it - no change.
We should simply remove this entire if and just keep the one below which uses default binder if it's null.

Maybe keep the comments in some way.

Reply:
Right. I also think adding `IsDefault()` to the AssemblyBinder could simplify a few places that do `if (binder == GetDomain()->GetBinder()) { . . . }`